src/Hit/Git.hs

```diff
@@ -42,21 +42,28 @@ runFresh (nameOrMaster -> branch) = do
     "git" ["rebase", "origin/" <> branch]
 
 -- | @hit new@ command.
-runNew :: Int -> IO ()
-runNew issueNum = do
+runNew :: Int -> Bool -> IO ()
+runNew issueNum assignOwner = do
     login <- getUsername
     let issueId = mkIssueId issueNum
-    issueTitle <- getIssueTitle issueId
-    let shortDesc = mkShortDesc issueTitle
+    issue <- fetchIssueStrict issueId
+    let shortDesc = mkShortDesc (issueTitle issue)
     let branchName = login <> "/" <> show issueNum <> "-" <> shortDesc
     "git" ["checkout", "-b", branchName]
+    when assignOwner (attemptAssignment issue)
   where
     mkShortDesc :: Text -> Text
     mkShortDesc =
           T.intercalate "-"
         . take 5
         . words
         . T.filter (\c -> isAlphaNum c || isDigit c || isSpace c)
+    attemptAssignment :: Issue -> IO ()
+    attemptAssignment = either printError finish <=< assignToIssue
```

Comment:
I feel like here you shouldn't try to handle any errors and print stuff. The `assignToIssue` function should use `errorMessage` and `exitFailure` in case of fails.

Reply:
The [assignAction](https://github.com/StevenXL/hit-on/blob/a2fe9f891998d92cdfe090be06d73114f07e1197/src/Hit/Issue.hs#L174) seems very verbose to me. Any way to clean it up while still being able to differentiate between the case where the user was already assigned to the issue and the case where the user was newly assigned to issue?